docs/sphinx/cookbook/shared_memory.rst

```diff
@@ -0,0 +1,73 @@
+.. _shared_memory:
+
+=======================
+Using IPC Shared Memory 
+=======================
+
+Umpire supports the use of IPC Shared Memory on the HOST memory resource. To do this, ``UMPIRE_ENABLE_IPC_SHARED_MEMORY`` 
+should be set to ``On``. Note that you can use IPC Shared Memory with MPI enabled or disabled.
+
+First, to get started with the shared memory allocator, set up the traits. For example:
+
+.. code-block:: cpp
+
+    auto traits{umpire::get_default_resource_traits("SHARED")};
+
+The ``traits`` above is a struct of different properties for your shared allocator. You can
+set the maximum size of the allocator with ``traits.size`` and set the scope of the allocator.
+
+For example, you can set the scope to socket:
+
+.. code-block:: cpp
+
+   traits.scope = umpire::MemoryResourceTraits::shared_scope::socket;
+
+However, by default the scope will be set to "node".
+
+Next, create the shared memory allocator:
+
+.. code-block:: cpp
+
+   auto node_allocator{rm.makeResource("SHARED::node_allocator", traits)};
+
+.. note::
+   The name of the Shared Memory allocators MUST have "SHARED" in the name. This will help
+   Umpire distinguish the allocators as Shared Memory allocators specifically.
+
+Now you can allocate and deallocate shared memory with:
+
+.. code-block:: cpp
+
+   void* ptr{node_allocator.allocate("allocation_name_2", sizeof(uint64_t))};
+   ...
+   node_allocator.deallocate(ptr);
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Important Notes About Shared Memory
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Because we are dealing with shared memory there are a few unique characteristics of the Shared Memory allocators
+which set it apart from other Umpire allocators.
+
+1. Once you allocate shared memory, that block of memory is fixed. If you need a bigger size, you will have to create a new one.
+2. If you want to see how much memory is available for a shared memory allocator, use the ``getActualSize()`` function.
+3. File descriptors are used for the shared memory. These files will be under ``/dev/shm``.
+
```

Comment:
Thanks @kab163  for adding documentation for this.  I'm glad that interest seems to be building and documentation is needed.  

You are correct in your statement made earlier that Umpire does not need to have MPI enabled in order to provide users with IPC Shared Memory.  HOWEVER, if they wish to associated shared memory with MPI communicators, I believe that Umpire will need to also have MPI set to On when it is compiled.  I could be wrong here, but this is how I remember it.  @davidbeckingsale might be able to shed further light or correct me if I am wrong.